c2rust-transpile/tests/snapshots/exprs.c

```diff
@@ -0,0 +1,33 @@
+int side_effect(int* x){
+    *x = 1;
+    return 10;
+}
+
+void unary_without_side_effect(){
+    int i = 5;
+    -i;
+    +i;
+    ~i;
+    !i;
+    &i;
+    *&i;
```

Comment:
So these unary ops now emit nothing, and while that's generally correct (I haven't thought through the exact semantics of the `*&` one yet), it may also delete code that was intentionally written in C, similar to #1326.  For things like `i++`, removing the `i;` and only keeping the `i += 1;` makes perfect sense, but I'm not sure for these other unary operators.

Reply:
ok, i force push new code that limit to not write unused code only when operator is increment/decrement